pkg/datapath/loader/loader.go

```diff
@@ -135,15 +170,35 @@ func nullifyStringSubstitutions(strings map[string]string) map[string]string {
 	return nullStrings
 }
 
+func (l *loader) bpfMasqAddrs(ifName string) (masq4, masq6 netip.Addr) {
+	if l.cfg.DeriveMasqIPAddrFromDevice != "" {
+		ifName = l.cfg.DeriveMasqIPAddrFromDevice
+	}
+
+	iter, _ := l.nodeAddrs.Get(
+		l.db.ReadTxn(),
+		tables.NodeAddressDeviceNameIndex.Query(ifName),
+	)
+	for addr, _, ok := iter.Next(); ok; addr, _, ok = iter.Next() {
+		if !addr.Primary {
+			continue
+		}
+		if addr.Addr.Is4() && masq4.IsUnspecified() {
+			masq4 = addr.Addr
```

Comment:
This implicitly causes this function to return the last address from the StateDB query. Is this the intended behavior? If so there should at least be a comment here explaining why that's the case.

Reply:
Doesn't it return the first addr? After first hit, the `masq4` is set, and then the setting is skipped for subsequent iterations.

Anyway, I think you might have scratched a subtle change in the addr selection. Previously, if the `firstGlobalAddr()` didn't find an IP addr for the given iface, then some a public IP addr of some other device was returned:
```
	        // Just make sure that we always return the same one and not a
		// random one. More info in the issue GH-7637.
		sort.SliceStable(ipsPublic, func(i, j int) bool {
			return ipsPublic[i].LinkIndex < ipsPublic[j].LinkIndex
		})
```

I wonder whether that behavior is still needed.

Reply:
Added a fallback to pick some public IP :). PTAL and review the new code carefully.

Reply:
Thanks! That code is needed for BPF NodePort too. Let's move it to `pkg/node/address.go`?

Reply:
Reworked it so that now we have a "wildcard device" (`*`) in the NodeAddress table which looks at all devices to find a suitable, preferably public IP.